File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Examinations/ExaminationsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ExaminationsRow } from "./ExaminationsRow";

export interface ExaminationsColumns {
    ExaminationId: Column<ExaminationsRow>;
    AppointmentId: Column<ExaminationsRow>;
    DoctorName: Column<ExaminationsRow>;
    PatientName: Column<ExaminationsRow>;
    Complaint: Column<ExaminationsRow>;
    ExaminationDate: Column<ExaminationsRow>;
    DoctorNote: Column<ExaminationsRow>;
}

export class ExaminationsColumns extends ColumnsBase<ExaminationsRow> {
    static readonly columnsKey = 'Examinations.Examinations';
    static readonly Fields = fieldsProxy<ExaminationsColumns>();
}

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Examinations/Examinations/ExaminationsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace MuayeneYonetimPortali.Examinations.Columns;

[ColumnsScript("Examinations.Examinations")]
[BasedOnRow(typeof(ExaminationsRow), CheckNames = true)]
public class ExaminationsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId")]
    public int ExaminationId { get; set; }
    [EditLink]
    public int AppointmentId { get; set; }
    [EditLink]
    public string DoctorName { get; set; }
    [EditLink]
    public string PatientName { get; set; }
    [EditLink]
    public string Complaint { get; set; }
    [EditLink]
    public DateTime ExaminationDate { get; set; }
    [EditLink]
    public string DoctorNote { get; set; }
}